ui-react/packages/atlasmap/src/UI/ConditionalExpressionInput.tsx

```diff
@@ -1,30 +1,106 @@
 import React, { FunctionComponent } from "react";
-import { ToolbarGroup, ToolbarItem, Button } from "@patternfly/react-core";
-import { css, StyleSheet } from "@patternfly/react-styles";
 import {
-  // ExpressionContent,
-  IExpressionContentProps,
-} from "./ExpressionContent";
+  ToolbarGroup,
+  ToolbarItem,
+  Button,
+  Tooltip,
+} from "@patternfly/react-core";
+import { css, StyleSheet } from "@patternfly/react-styles";
+import { ExpressionContent } from "./ExpressionContent";
 
 const styles = StyleSheet.create({
   toolbarItem: { flex: 1 },
 });
+export interface IConditionalExpressionInputProps {
+  executeFieldSearch: (searchFilter: string, isSource: boolean) => string[][];
+  mappingExpressionAddField: (
+    selectedField: string,
+    newTextNode: any,
+    atIndex: number,
+    isTrailer: boolean,
+  ) => void;
+  mappingExpressionClearText: (
+    nodeId?: string,
+    startOffset?: number,
+    endOffset?: number,
+  ) => any;
+  isMappingExpressionEmpty: boolean;
+  mappingExpressionInit: () => void;
+  mappingExpressionInsertText: (
+    str: string,
+    nodeId?: string | undefined,
+    offset?: number | undefined,
+  ) => void;
+  mappingExpressionObservable: () => any;
+  mappingExpressionRemoveField: (
+    tokenPosition?: string,
+    offset?: number,
+    removeNext?: boolean,
+  ) => void;
+  onConditionalMappingExpressionEnabled: () => boolean;
+  onToggleExpressionMode: () => void;
+  mappingExpression?: string;
+  trailerId: string;
+}
+
+export const ConditionalExpressionInput: FunctionComponent<IConditionalExpressionInputProps> = ({
+  executeFieldSearch,
+  mappingExpressionAddField,
+  mappingExpressionClearText,
+  isMappingExpressionEmpty,
+  mappingExpressionInit,
+  mappingExpressionInsertText,
+  mappingExpressionObservable,
+  mappingExpressionRemoveField,
+  mappingExpression,
+  onConditionalMappingExpressionEnabled,
+  onToggleExpressionMode,
+  trailerId,
+}) => {
+  function onToggle() {
```

Comment:
This function does nothing useful, it's wrapping a callback without adding any value to it. Also the callback is already called at line #99, so there are two ways to do the same thing.

Reply:
yep - I've removed onToggleExpressionMode from the EclipseContent props